hera_cal/tests/test_apply_cal.py

```diff
@@ -42,6 +42,77 @@ def test_check_polarization_consistency(self):
         with pytest.raises(KeyError):
             ac._check_polarization_consistency(data, gains)
 
+    def test_build_gains_by_cadences(self):
+        # test upsampling
+        data = {(0, 1, 'nn'): np.ones((8, 10), dtype=complex)}
+        gains = {(0, 'Jnn'): np.array([np.arange(10)]).repeat(2, axis=0).astype(complex)}
+        flags = {(0, 'Jnn'): np.zeros((2, 10), dtype=bool)}
+        gains_by_Nt, cal_flags_by_Nt = ac.build_gains_by_cadences(data, gains, cal_flags=flags)
+        for Nt in [2, 4, 8]:
```

Comment:
I think it'd be nice to add testing for non-power-of-two times, once we add the appropriate error/warning handling above.

Reply:
I've added warnings and tests for those warnings